### ISSUE TITLE
[Discussion] How can we QA our products better?

### ISSUE THREAD

**Issue by aroltsch** [department-of-veterans-affairs/caseflow#6824]
We've had situations where we've piloted a feature without fully testing it. We have Capybara automated tests but it is not always enough. What can we do to improve our user experience and catch bugs before we pilot?

1. When we are ready to pilot a feature, should we dedicate time for a few people (2 designers, 2 engineers) to test end to end user experience?
2. Each time we pilot a feature, we can choose one engineer to test the feature? 
3. Should we work on writing more Capybara tests?

**Comment by laurjpeterson** [comment]
Thanks for writing this up! I agree with #1 here, in order to catch not only bugs but also potential gaps in meeting business requirements. We could add that this should precede testing with Jed.

Would this QA happen in UAT or demo?

**Comment by nicholasholtz** [comment]
Just for alternative views, this is the Intake validation process: https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Intake/Validation.md

Specifically worth noting is item 4 in the validation process: 
> During weekly planning meeting, closed validation tickets are reviewed and the "sierra" label removed so we can keep track of our weekly progress by looking at the done column in waffle.

**Comment by lowellrex** [comment]
Whenever we pilot a feature with a group of users I think it should be as small a group as possible (2? 3?) and we should check in with every one of our initial group of pilot users every single day (preferably in person so we can identify ways they are using the feature we did not expect and that they may not be aware of enough to mention in an skype message or email).

We should certainly be trying to catch as many bugs as we can pre-release, but I think we need to build relationships with our alpha release testers and rely on them to identify bugs that they experience in real-world usage so we can quickly patch them.

**Comment by amprokop** [comment]
@lowellrex's suggestion sounds pretty similar to the efforts @laurjpeterson already leads, correct me plz if not.

A validation column might be good as well.

**Comment by lowellrex** [comment]
@amprokop I guess what is different with what I am suggesting is that the programmers be the ones checking in with our alpha testers. And that they do it every day. Maybe this is already happening, but I have never once had a face-to-face with an alpha tester.

**Comment by laurjpeterson** [comment]
For attorney check out, we did start with 3 users and designers and I checked in with them (though not daily). For judge pilots, we were ambitious with a larger pilot group (24 users). Designers and I checked in and shadowed a few of them first, then we started our user feedback initiative because it wasn't enough, and to include engineers as well. 

I like @lowellrex's idea for engineers to sit with our small pool of users post-release, especially for co-located pilot (2 co-located, 10 attorneys). We did this with Certification pilots last summer. 

I'm also down to try an In-Validation column as well!
